utils/cli/core/core.py

```diff
@@ -77,15 +81,23 @@ def tasks_delete(self, task_ids, **kwargs):
                 else:
                     raise e
 
-    def tasks_frame(self, task_id, frame_ids, outdir='', **kwargs):
+    def tasks_frame(self, task_id, frame_ids, outdir='', quality='original', **kwargs):
         """ Download the requested frame numbers for a task and save images as
         task_<ID>_frame_<FRAME>.jpg."""
         for frame_id in frame_ids:
-            url = self.api.tasks_id_frame_id(task_id, frame_id)
+            url = self.api.tasks_id_frame_id(task_id, frame_id, quality)
             response = self.session.get(url)
             response.raise_for_status()
             im = Image.open(BytesIO(response.content))
-            outfile = 'task_{}_frame_{:06d}.jpg'.format(task_id, frame_id)
+            mime_type = im.get_format_mimetype() or 'image/jpg'
+            im_ext = mimetypes.guess_extension(mime_type)
+            # FIXME It is better to use meta information from the server
```

Comment:
This information can also be obtained from the data header - most image formats have this. I think the extension depends on the use - for compressed you'd use `.jpg` and for raw you'd request this from the server.